### DISCUSSION TITLE
英文文献两个作者，引用时如何出现 and？

### DISCUSSION THREAD

**Discussion by iamforerunner** [stone-zeng/fduthesis#301] [Q&A] [answered]
英文文献两个作者时，引用结果是 Zhang, Wang, 2020。目标结果是 Zhang and Wang, 2018

这段代码用不了了
\DefineBibliographyStrings{english}{ % 针对英文两个作者，弄出 Zhang and Wang, 2020，而非 Zhang, Wang, 2020
andincite = { and },
andincitecn = {和},
andothersincite = {et al.\isdot},
% andothersincitecn = {等{\adddot}} %adddot才能避开标点追踪
andothersincitecn = {等}
}

**Comment by sikouhjw** [comment]
1. 你提问的代码是『你』自己回答的代码[^1]
2. 你给的代码是可以起作用的
```tex
\begin{filecontents}{xampl-zh.bib}
@article{desmarais1992carbon,
  author       = {Des Marais, David J and Strauss, H and Summons, R E and
                 others},
  title        = {Carbon Isotope Evidence for the Stepwise Oxidation of the
                 Proterozoic Environment},
  journal      = {Nature},
  year         = {1992},
  volume       = {359},
  pages        = {605--609},
}

@article{saito2006jadeite,
  author       = {Saito, M and Miyazaki, K},
  title        = {Jadeite-bearing Metagabbro in Serpentinite Mélange of the
                 {"Kurosegawa Belt" in Izumi Town, Yatsushiro City, Kumamoto
                 Prefecture, central Kyushu}},
  journal      = {Bulletin of the geological survey of Japan},
  year         = {2006},
  volume       = {57},
  number       = {5/6},
  pages        = {169--176},
}
\end{filecontents}

\documentclass{ctexart}
\usepackage[backend=biber,style=gb7714-2015,maxcitenames=2]{biblatex}
\DefineBibliographyStrings{english}{
  andincite = { and },
}
\addbibresource{xampl-zh.bib}
\begin{document}
\citet{desmarais1992carbon,saito2006jadeite}
\end{document}
```
![图片](https://user-images.githubusercontent.com/49770925/236263767-100d2a28-afc0-4d56-b8ce-8d9a60eec146.png)

[^1]: https://github.com/stone-zeng/fduthesis/discussions/255#discussioncomment-5787478

**Reply by iamforerunner** [reply]
> 1. 你提问的代码是『你』自己回答的代码[1](#user-content-fn-1-409da6051b1c9b0636afcb8ab98be82c)
> 2. 你给的代码是可以起作用的
> 
> ```
> \begin{filecontents}{xampl-zh.bib}
> @article{desmarais1992carbon,
>   author       = {Des Marais, David J and Strauss, H and Summons, R E and
>                  others},
>   title        = {Carbon Isotope Evidence for the Stepwise Oxidation of the
>                  Proterozoic Environment},
>   journal      = {Nature},
>   year         = {1992},
>   volume       = {359},
>   pages        = {605--609},
> }
> 
> @article{saito2006jadeite,
>   author       = {Saito, M and Miyazaki, K},
>   title        = {Jadeite-bearing Metagabbro in Serpentinite Mélange of the
>                  {"Kurosegawa Belt" in Izumi Town, Yatsushiro City, Kumamoto
>                  Prefecture, central Kyushu}},
>   journal      = {Bulletin of the geological survey of Japan},
>   year         = {2006},
>   volume       = {57},
>   number       = {5/6},
>   pages        = {169--176},
> }
> \end{filecontents}
> 
> \documentclass{ctexart}
> \usepackage[backend=biber,style=gb7714-2015,maxcitenames=2]{biblatex}
> \DefineBibliographyStrings{english}{
>   andincite = { and },
> }
> \addbibresource{xampl-zh.bib}
> \begin{document}
> \citet{desmarais1992carbon,saito2006jadeite}
> \end{document}
> ```
> 
> ![图片](https://user-images.githubusercontent.com/49770925/236263767-100d2a28-afc0-4d56-b8ce-8d9a60eec146.png)
> 
> ## Footnotes
> 1. https://github.com/stone-zeng/fduthesis/discussions/255#discussioncomment-5787478 [↩](#user-content-fnref-1-409da6051b1c9b0636afcb8ab98be82c)

在fduthesis里报错啊
Undefined control sequence.

**Comment by note286** [comment, marked as answer]
```latex
\PassOptionsToPackage{maxcitenames=2}{biblatex}
\documentclass{fduthesis}
\fdusetup{style/bib-resource=reference.bib}
\AtBeginEnvironment{document}{\DefineBibliographyStrings{english}{andincite={\space and\space}}}
\begin{document}
\citet{desmarais1992carbon,saito2006jadeite}
\end{document}
```


![image](https://user-images.githubusercontent.com/16207292/236360448-f51a6e7f-878d-45f2-ab6c-f124df950ba1.png)
